datalad/core/local/diff.py

```diff
@@ -97,6 +97,25 @@ class Diff(Interface):
             constraints=EnsureStr() | EnsureNone()),
     )
 
+    _examples_ = [
+        dict(text="""Show unsaved changes in the current dataset""",
+             code_py="diff(dataset='.')",
+             code_cmd="datalad diff"),
+        dict(text="""Compare a previous dataset state identified by shasum
```

Comment:
The triple-quote,  multiliine string here and the other instances throughout these commits aren't rendered correctly.  For example, here is what I see in the output of `datalad diff --help`:

```
Compare a previous dataset state identified by shasum
Against current worktree::

   % datalad diff --from <SHASUM>
```

Reply:
This smells like a bug in the general docstring formatting code, I didn't spot anything specific to examples....

Reply:
The upper-casing is certainly a bug, and perhaps not stripping the newline between "shasum" and "against" is.  While it's outside this PR's scope to fix that, I think it should avoid a form that introduces lots of these misrendered cases.  Using single quotes, for example, would fix it.

Reply:
Thanks, I will use single quotes.

Reply:
> The upper-casing is certainly a bug, 

That's fixed with gh-4094.

> and perhaps not stripping the newline between "shasum" and "against" is

@adswa  Given the `text` field of the example is free form, do you think we should do something like this?

```diff
diff --git a/datalad/interface/base.py b/datalad/interface/base.py
index 15fe8850c..9f6798785 100644
--- a/datalad/interface/base.py
+++ b/datalad/interface/base.py
@@ -402,7 +402,7 @@ def build_example(example, api='python'):
     if code_field not in example:
         # only show an example if it exist for the API
         return ''
-    description = dedent_docstring(example.get('text'))
+    description = textwrap.fill(example.get('text'))
     # this indent the code snippet to get it properly rendered as code
     # we are not using textwrap.fill(), because it would not acknowledge
     # any meaningful structure/formatting of code snippets. Instead, we
```

Reply:
It makes sense to me, yes.